android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/config/MapControls.kt

```diff
@@ -1,34 +1,97 @@
 package com.stadiamaps.ferrostar.maplibreui.config
 
-import android.view.Gravity
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.WindowInsets
+import androidx.compose.foundation.layout.asPaddingValues
+import androidx.compose.foundation.layout.calculateEndPadding
+import androidx.compose.foundation.layout.systemBars
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import com.maplibre.compose.runtime.localLayoutDirection
 import com.maplibre.compose.settings.AttributionSettings
 import com.maplibre.compose.settings.CompassSettings
 import com.maplibre.compose.settings.LogoSettings
+import com.maplibre.compose.settings.MapControlPosition
 import com.maplibre.compose.settings.MapControls
-import com.maplibre.compose.settings.MarginInsets
+import com.stadiamaps.ferrostar.composeui.runtime.paddingForGridView
 
 /**
  * Returns the map controls for the given configuration.
  *
- * @param isLandscape whether the NavigationView is in landscape orientation.
- * @param isArrivalExpanded weather the arrival view is expanded.
+ * @param arrivalViewHeight The height of the arrival view.
+ * @param horizontalPadding The horizontal padding to apply to the map controls. Defaults to 16.dp
+ *   to match the default padding of the arrival view.
+ * @param verticalPadding The vertical padding to apply to the map controls from the top of the
+ *   arrival view. Defaults to 8.dp.
+ *
+ * TODO: This function is attempting to optimize the map controls for many screen sizes, system
+ *   bars, and orientations. We should remain open to feedback for specific cases.
  */
-internal fun mapControlsFor(isLandscape: Boolean, isArrivalExpanded: Boolean): MapControls {
-  val bottom =
-      when {
-        isLandscape -> 32
-        isArrivalExpanded -> 32 * 9
-        else -> 32 * 7
-      }
-
-  return MapControls(
-      attribution =
-          AttributionSettings(
-              gravity = Gravity.BOTTOM or Gravity.END,
-              margins = MarginInsets(end = 270, bottom = bottom)),
-      compass = CompassSettings(enabled = false),
-      logo =
-          LogoSettings(
-              gravity = Gravity.BOTTOM or Gravity.END,
-              margins = MarginInsets(end = 32, bottom = bottom)))
+@Composable
+internal fun rememberMapControlsFor(
+    arrivalViewHeight: Dp = 0.dp,
+    horizontalPadding: Dp = 16.dp,
+    verticalPadding: Dp = 8.dp
+): State<MapControls> {
```

Comment:
This is a valid discussion and maybe the best move is we push this function to Maplibre-Compose as something more generic. It might take a couple of passes, but I think there's value in `rememberMapControls` that allows dynamically triggering padding within the window insets (or inside another calculated "window") 

Bonus is this would let us debug that crash locally in the maplibre-compose example app.

Reply:
> maybe the best move is we push this function to Maplibre-Compose as something more generic

Agreed. I think that's probably the right place for this.

> Bonus is this would let us debug that crash locally in the maplibre-compose example app.

Yeah, actually I think it probably would :D